src/tls13.c

```diff
@@ -7970,6 +7970,24 @@ int wolfSSL_update_keys(WOLFSSL* ssl)
     return ret;
 }
 
+/* Whether a response is waiting for key update request.
+ *
+ * ssl        The SSL/TLS object.
+ * required   0 when no key update response required.
+ *            1 when no key update response required.
+ * return  0 on success.
+ * return  BAD_FUNC_ARG when ssl is NULL or not using TLS v1.3
+ */
+int wolfSSL_key_update_response(WOLFSSL* ssl, int* required)
```

Comment:
I'm curious why can't the `ssl->keys.updateResponseReq` be the return value? The error condition could be `<0` and this would eliminate the complexity of an additional parameter.

Reply:
Good rule of API design is that you don't have two things being passed through one variable/return code.

OpenSSL was a great example of where this went wrong.
The code had numerous problems, that have now been mostly/all fixed, where the return value was negative on failure and 0/1/+ve on success.
Some calling code would check for '!= 0' instead of '< 0' for failure. The code was silently failing when a positive value was returned. This actually caused security errors like not properly detecting certificate validation errors.